device-telemetry/app/com/microsoft/azure/iotsolutions/devicetelemetry/services/models/actions/EmailAction.java

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+package com.microsoft.azure.iotsolutions.devicetelemetry.services.models.actions;
+
+import com.fasterxml.jackson.databind.PropertyNamingStrategy;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonNaming;
+import com.microsoft.azure.iotsolutions.devicetelemetry.services.exceptions.InvalidInputException;
+import play.libs.Json;
+
+import javax.mail.internet.InternetAddress;
+import java.util.*;
+
+@JsonDeserialize(as = EmailAction.class)
+@JsonNaming(PropertyNamingStrategy.UpperCamelCaseStrategy.class)
+public final class EmailAction implements IAction {
+
+    private ActionType type;
+    private Map<String, Object> parameters;
+
+    private static final String SUBJECT = "Subject";
+    private static final String NOTES = "Notes";
+    private static final String RECIPIENTS = "Recipients";
+
+    public EmailAction() {
+        this.type = ActionType.Email;
+        this.parameters = new TreeMap<String, Object>(String.CASE_INSENSITIVE_ORDER) {{
+            put(NOTES, "");
+        }};
+    }
+
+    public EmailAction(Map<String, Object> parameters) throws InvalidInputException {
+        this(ActionType.Email, parameters);
+    }
+
+    public EmailAction(ActionType type, Map<String, Object> parameters) throws InvalidInputException {
```

Comment:
parameters and this.parameters is probably not the best distinguishing name here. It is a common pattern for getters and setters but it is pretty simple to mess something up unexpectedly by including or forgetting to include "this" somewhere.